# Comparing Bitcoin and Ethereum Data
#### This project will be comparing both trends of the cryptocurrencies and individually analysed
This project is a self made project by Gordon Wu


The data is taken from https://www.cryptodatadownload.com

In [1]:
import numpy as np
import pandas as pd
import os

#### Retrieving database

In [2]:
BTCdf = pd.read_csv('data/BitcoinHourly.csv')
ETHdf = pd.read_csv('data/ETHHourly.csv')

### Checking database contents

In [3]:
BTCdf

unix                 date   symbol      open      high       low  \
0      1630108800  2021-08-28 00:00:00  BTC/USD  49100.49  49250.00  48875.00   
1      1630105200  2021-08-27 23:00:00  BTC/USD  49066.42  49139.95  48906.19   
2      1630101600  2021-08-27 22:00:00  BTC/USD  48999.57  49069.38  48725.00   
3      1630098000  2021-08-27 21:00:00  BTC/USD  48995.23  49089.58  48889.00   
4      1630094400  2021-08-27 20:00:00  BTC/USD  48332.44  49187.00  48188.88   
...           ...                  ...      ...       ...       ...       ...   
28814  1526378400  2018-05-15 10:00:00  BTC/USD   8708.32   8865.00   8695.11   
28815  1526374800  2018-05-15 09:00:00  BTC/USD   8728.49   8754.40   8701.35   
28816  1526371200  2018-05-15 08:00:00  BTC/USD   8739.00   8750.27   8660.53   
28817  1526367600  2018-05-15 07:00:00  BTC/USD   8740.99   8766.00   8721.11   
28818  1526364000  2018-05-15 06:00:00  BTC/USD   8733.86   8796.68   8707.28   

          close    Volume BTC    Volume USD  
0      48992.83  3.633715e+01  1.780260e+06  
1      49100.49  5.579023e+01  2.739328e+06  
2      49014.60  3.628698e+01  1.778592e+06  
3      49005.09  5.816773e+01  2.850515e+06  
4      48950.41  1.283265e+02  6.281634e+06  
...         ...           ...           ...  
28814   8795.90  1.110127e+07  1.260690e+03  
28815   8708.32  1.593992e+06  1.826200e+02  
28816   8728.49  7.986063e+06  9.177900e+02  
28817   8739.00  2.390399e+06  2.735800e+02  
28818   8740.99  4.906603e+06  5.599300e+02  

[28819 rows x 9 columns]

In [4]:
ETHdf

unix                 date   symbol     open     high      low  \
0      1630108800  2021-08-28 00:00:00  ETH/USD  3271.67  3288.71  3260.01   
1      1630105200  2021-08-27 23:00:00  ETH/USD  3276.08  3283.47  3263.28   
2      1630101600  2021-08-27 22:00:00  ETH/USD  3276.46  3278.17  3255.52   
3      1630098000  2021-08-27 21:00:00  ETH/USD  3264.84  3280.00  3261.52   
4      1630094400  2021-08-27 20:00:00  ETH/USD  3235.00  3277.00  3225.46   
...           ...                  ...      ...      ...      ...      ...   
28814  1526378400  2018-05-15 10:00:00  ETH/USD   727.32   739.37   725.04   
28815  1526374800  2018-05-15 09:00:00  ETH/USD   731.68   731.82   726.98   
28816  1526371200  2018-05-15 08:00:00  ETH/USD   733.82   734.92   721.58   
28817  1526367600  2018-05-15 07:00:00  ETH/USD   733.00   735.99   730.45   
28818  1526364000  2018-05-15 06:00:00  ETH/USD   732.10   736.67   730.94   

         close    Volume ETH    Volume USD  
0      3268.53  2.095992e+02  6.850812e+05  
1      3276.68  2.336721e+02  7.656687e+05  
2      3276.07  3.793405e+02  1.242746e+06  
3      3276.46  4.899050e+02  1.605154e+06  
4      3263.73  1.705007e+03  5.564682e+06  
...        ...           ...           ...  
28814   733.59  1.424388e+06  1.943150e+03  
28815   727.32  4.384113e+05  6.010300e+02  
28816   731.68  2.860231e+06  3.924550e+03  
28817   733.82  2.594569e+05  3.538200e+02  
28818   733.00  7.662087e+05  1.043910e+03  

[28819 rows x 9 columns]

### Checking the columns

In [5]:
BTCdf_cols = BTCdf.columns
BTCdf_cols

Index(['unix', 'date', 'symbol', 'open', 'high', 'low', 'close', 'Volume BTC',
       'Volume USD'],
      dtype='object')

In [6]:
ETHdf_cols = ETHdf.columns
ETHdf_cols

Index(['unix', 'date', 'symbol', 'open', 'high', 'low', 'close', 'Volume ETH',
       'Volume USD'],
      dtype='object')

### Comparing the % of times where the opening price is less than the closing price

In [7]:
BTCdf[BTCdf.open < BTCdf.close].size/BTCdf.size *100

50.94902668378501

In [8]:
ETHdf[ETHdf.open < ETHdf.close].size/ETHdf.size *100

50.33831847045352

Evidently, Bitcoin has more increases in prices in the hourly periods but it is negligible
### Comparing the % of times when the closing prices is greater than 101.5% of the opening price
Creating a new column which compares the opening and closing prices % wise

In [9]:
BTCdf['openclose_pct'] = ((BTCdf['close'] - BTCdf['open'])/BTCdf['open']) * 100
BTCdf

unix                 date   symbol      open      high       low  \
0      1630108800  2021-08-28 00:00:00  BTC/USD  49100.49  49250.00  48875.00   
1      1630105200  2021-08-27 23:00:00  BTC/USD  49066.42  49139.95  48906.19   
2      1630101600  2021-08-27 22:00:00  BTC/USD  48999.57  49069.38  48725.00   
3      1630098000  2021-08-27 21:00:00  BTC/USD  48995.23  49089.58  48889.00   
4      1630094400  2021-08-27 20:00:00  BTC/USD  48332.44  49187.00  48188.88   
...           ...                  ...      ...       ...       ...       ...   
28814  1526378400  2018-05-15 10:00:00  BTC/USD   8708.32   8865.00   8695.11   
28815  1526374800  2018-05-15 09:00:00  BTC/USD   8728.49   8754.40   8701.35   
28816  1526371200  2018-05-15 08:00:00  BTC/USD   8739.00   8750.27   8660.53   
28817  1526367600  2018-05-15 07:00:00  BTC/USD   8740.99   8766.00   8721.11   
28818  1526364000  2018-05-15 06:00:00  BTC/USD   8733.86   8796.68   8707.28   

          close    Volume BTC    Volume USD  openclose_pct  
0      48992.83  3.633715e+01  1.780260e+06      -0.219265  
1      49100.49  5.579023e+01  2.739328e+06       0.069436  
2      49014.60  3.628698e+01  1.778592e+06       0.030674  
3      49005.09  5.816773e+01  2.850515e+06       0.020124  
4      48950.41  1.283265e+02  6.281634e+06       1.278582  
...         ...           ...           ...            ...  
28814   8795.90  1.110127e+07  1.260690e+03       1.005705  
28815   8708.32  1.593992e+06  1.826200e+02      -0.231082  
28816   8728.49  7.986063e+06  9.177900e+02      -0.120265  
28817   8739.00  2.390399e+06  2.735800e+02      -0.022766  
28818   8740.99  4.906603e+06  5.599300e+02       0.081636  

[28819 rows x 10 columns]

In [10]:
ETHdf['openclose_pct'] = ((ETHdf['close'] - ETHdf['open'])/ETHdf['open']) * 100
ETHdf

unix                 date   symbol     open     high      low  \
0      1630108800  2021-08-28 00:00:00  ETH/USD  3271.67  3288.71  3260.01   
1      1630105200  2021-08-27 23:00:00  ETH/USD  3276.08  3283.47  3263.28   
2      1630101600  2021-08-27 22:00:00  ETH/USD  3276.46  3278.17  3255.52   
3      1630098000  2021-08-27 21:00:00  ETH/USD  3264.84  3280.00  3261.52   
4      1630094400  2021-08-27 20:00:00  ETH/USD  3235.00  3277.00  3225.46   
...           ...                  ...      ...      ...      ...      ...   
28814  1526378400  2018-05-15 10:00:00  ETH/USD   727.32   739.37   725.04   
28815  1526374800  2018-05-15 09:00:00  ETH/USD   731.68   731.82   726.98   
28816  1526371200  2018-05-15 08:00:00  ETH/USD   733.82   734.92   721.58   
28817  1526367600  2018-05-15 07:00:00  ETH/USD   733.00   735.99   730.45   
28818  1526364000  2018-05-15 06:00:00  ETH/USD   732.10   736.67   730.94   

         close    Volume ETH    Volume USD  openclose_pct  
0      3268.53  2.095992e+02  6.850812e+05      -0.095975  
1      3276.68  2.336721e+02  7.656687e+05       0.018315  
2      3276.07  3.793405e+02  1.242746e+06      -0.011903  
3      3276.46  4.899050e+02  1.605154e+06       0.355913  
4      3263.73  1.705007e+03  5.564682e+06       0.888099  
...        ...           ...           ...            ...  
28814   733.59  1.424388e+06  1.943150e+03       0.862069  
28815   727.32  4.384113e+05  6.010300e+02      -0.595889  
28816   731.68  2.860231e+06  3.924550e+03      -0.291625  
28817   733.82  2.594569e+05  3.538200e+02       0.111869  
28818   733.00  7.662087e+05  1.043910e+03       0.122934  

[28819 rows x 10 columns]

In [11]:
ETHdf[ETHdf.openclose_pct > 1.5].size/ETHdf.size *100

4.913425170894202

In [12]:
BTCdf[BTCdf.openclose_pct > 1.5].size/BTCdf.size *100

2.7551268260522574

Evidently from this outcome, we can imply that Ethereum has a higher overall amount of volume which had a significant increase in prices.

### Comparing the % of times when the closing prices is less than 98.5% of the opening price

In [13]:
ETHdf[ETHdf.openclose_pct < -1.5].size/ETHdf.size *100

4.701759256046358

In [14]:
BTCdf[BTCdf.openclose_pct < -1.5].size/BTCdf.size *100

2.876574482112495

We can see that Ethereum has a higher portion of hourly periods that drop below 1.5% from the closing rate so we can assume that even though Ethereum has a higher amount of periods which increase in prices, it also has a larger portion which drops significantly, causing it to be riskier to invest.

### Comparing the volatility of both cryptocurrencies

In [15]:
ETHdf.openclose_pct.std()

1.0453170988531089

In [16]:
BTCdf.openclose_pct.std()

0.8304711696806759

This supports the statement above highlighting how the volatility in Ethereum is higher compared to Bitcoin, and so it is riskier to go into Ethereum yet it may be more rewarding

In [17]:
ETHdf.openclose_pct.mean()

0.008907173516535749

In [18]:
BTCdf.openclose_pct.mean()

0.009460785221856572

### Percent of times that both crypto increase/decrease together
This allows us to determine how often the cryptocurrencies correlate with each other

In [19]:
# Creating a new column to see whether the cryptocurrency increased or not
BTCdf['increased'] = BTCdf['openclose_pct'] > 0
ETHdf['increased'] = ETHdf['openclose_pct'] > 0

# Creating a new table to compare if both has increased or not
df = {'correlates': []}
df = pd.DataFrame(df, columns= ['correlates'])
df['correlates'] = np.where(ETHdf['increased'] == BTCdf['increased'], 'True', 'False')
df.correlates.value_counts(normalize = True)

True     0.815677
False    0.184323
Name: correlates, dtype: float64

As it has an 81.57% correlation rate, it is evident that Bitcoin and Ethereum do follow similar trends

### The proportion of times that the cryptocurrencies follow the trend of the previous period

In [30]:
# Creating a new column which finds if the previous period has increased or not
BTCdf['increased (t-1)'] = BTCdf['increased'].shift(-1)

# Creating new database to find the trends
BTCdf_increased = {'follows': []}
BTCdf_increased = pd.DataFrame(BTCdf_increased, columns= ['follows'])
BTCdf_increased['follows'] = np.where(BTCdf['increased'] == BTCdf['increased (t-1)'], 'True', 'False')
BTCdf_increased.follows.value_counts(normalize = True)

False    0.54471
True     0.45529
Name: follows, dtype: float64

In [31]:
# Creating a new column which finds if the previous period has increased or not
ETHdf['increased (t-1)'] = ETHdf['increased'].shift(-1)

# Creating new database to find the trends
ETHdf_increased = {'follows': []}
ETHdf_increased = pd.DataFrame(ETHdf_increased, columns= ['follows'])
ETHdf_increased['follows'] = np.where(ETHdf['increased'] == ETHdf['increased (t-1)'], 'True', 'False')
ETHdf_increased.follows.value_counts(normalize = True)

False    0.541865
True     0.458135
Name: follows, dtype: float64

Evidently, from above we can tell that they do not follow the trend of the previous period and that it would be a failing strategy to do so, but yet, as seen above, as the correlation was between Ethereum and Bitcoin was high, we achieved similar results for both the trends on Ethereum and Bitcoin